tests/vm_test.go

```diff
@@ -1921,6 +1923,92 @@ status:
 			}, 2*time.Minute, 1*time.Second).Should(Succeed())
 		})
 	})
+
+	Context("when node becomes unhealthy", Serial, func() {
+		var nodeName string
+
+		AfterEach(func() {
+			node, err := kubevirt.Client().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			node.Spec.Unschedulable = false
+			node, err = kubevirt.Client().CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(node.Spec.Unschedulable).To(BeFalse())
+
+			libnode.UnTaint(nodeName, "node.kubernetes.io/unschedulable")
+		})
+		It("[Serial] the VMs running in that node should be respawned", func() {
+			var pod *k8sv1.Pod
+			var wg sync.WaitGroup
+
+			By("Starting VM")
+
+			vmi := libvmi.NewAlpine()
+			vmi.Namespace = testsuite.GetTestNamespace(vmi)
+			vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunning())
+
+			vm, err := kubevirt.Client().VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(beReady())
+			vmi = libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("Taint the node, it simulates that something went wrong in the cluster and workloads needs to be removed")
+
+			nodeName = vmi.Status.NodeName
+			libnode.Taint(nodeName, "node.kubernetes.io/unschedulable", k8sv1.TaintEffectNoExecute)
+
+			node, err := kubevirt.Client().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Disable scheduling in the node")
+
+			node.Spec.Unschedulable = true
+			node, err = kubevirt.Client().CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(node.Spec.Unschedulable).To(BeTrue())
+
+			By("VMi should be NotReady")
+
+			Eventually(ThisVMI(vmi), 300*time.Second, 2*time.Second).Should(HaveConditionFalse(v1.VirtualMachineInstanceReady))
+
+			By("Pods should be in terminating state")
+
+			Eventually(func() bool {
+				pod, err = libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+				return pod.DeletionTimestamp != nil
+			}, 30*time.Second, 2*time.Second).Should(BeTrue())
+
+			vmiWatch, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Watch(context.Background(), metav1.ListOptions{})
+
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				By("The VMi should transition to phase Failed")
+				Eventually(vmiWatch.ResultChan(), 60*time.Second).Should(Receive(WithTransform(func(e watch.Event) v1.VirtualMachineInstanceStatus {
```

Comment:
Is it necessary to do it with a watch? Wouldn't it be enough to poll with Eventually, or could you miss the phase change then?